React/Proptypes: Add additional video resource (#27366)
TheOdinProject/curriculum: react/the_react_ecosystem/type_checking_with_proptypes.md

### Assignment

<div class="lesson-content__panel" markdown="1">

1. Read through the [PropTypes documentation](https://reactjs.org/docs/typechecking-with-proptypes.html). It shows all of the types you can specify and some other useful things that can be done with it!
   - You may notice this resource suggests using TypeScript over PropTypes in modern React. This may well be true for production apps but for our purposes, we are solely interested in the concept of type checking within React, where TypeScript is out of scope.
1. You can even set up custom validators in PropTypes. For a more in-depth look into the benefits and use cases of PropTypes, read this comprehensive guide [Validating React Props with PropTypes on LogRocket](https://blog.logrocket.com/validate-react-props-proptypes/).

> 1. Dive into the nuanced comparison between PropTypes and TypeScript for type safety in React with [a detailed analysis found on StackOverflow](https://stackoverflow.com/questions/41746028/proptypes-in-a-typescript-react-application), highlighting key differences and use cases.

Reviewer: ```suggestion
1. Dive into the nuanced comparison between PropTypes and TypeScript for type safety in React with this [StackOverflow post on differences between PropTypes and TypeScript](https://stackoverflow.com/questions/41746028/proptypes-in-a-typescript-react-application).
```

Author: Hi @thatblindgeye, apologies for the delay, sure thing I will add these suggestions.

</div>

### Knowledge check

The following questions are an opportunity to reflect on key topics in this lesson. If you can't answer a question, click on it to review the material, but keep in mind you are not expected to memorize or master this knowledge.

- <a class="knowledge-check-link" href="#using-proptypes">How would we set up a basic implementation of PropTypes?</a>
- <a class="knowledge-check-link" href="#using-defaultprops">If we pass in a prop to a component that has a defaultProp defined, what would happen?</a>
- <a class="knowledge-check-link" href="https://stackoverflow.com/questions/41746028/proptypes-in-a-typescript-react-application">What is the difference between PropTypes and TypeScript?</a>

### Additional resources

This section contains helpful links to related content. It isn't required, so consider it supplemental.

> - If you find video lessons easier to absorb, consider watching [Web Dev Simplified's video on understanding PropTypes in React](https://www.youtube.com/watch?v=cx0S8JyiVxc). This video covers many of the topics discussed in this lesson.

Reviewer: Just wondering, does this video go into additional topics that this lesson doesn't cover, or is it just a video alternative to what this lesson presents?

Reviewer: @scottwright-dev depending on the answer to the above, I'm inclined to not include this video if it's just going over the same content in the lesson, just as a video. I feel like that's something a user could seek out on their own if they want a video.

Author: HI @thatblindgeye, it is essentially a summary of whats discussed in the lesson so feel free to omit.

Reviewer: Gotcha. I'd say let's revert this link back to the stock verbiage then.

Author: @thatblindgeye sure thing, all done!
